linkerd/app/inbound/src/direct.rs

```diff
@@ -148,29 +224,61 @@ where
         .into_inner()
 }
 
-// === impl HttpFromTransportHeader ===
-
-impl svc::Predicate<(TransportHeader, TcpAccept)> for HttpFromTransportHeader {
-    type Request = svc::Either<TcpEndpoint, (http::Version, TcpAccept)>;
-
-    // TODO: Use the transport header's `name` to inform gateway routing.
-    //
-    // TODO: Use the header's session protocol, i.e. for gateway processing.
-    fn check(&mut self, (h, mut t): (TransportHeader, TcpAccept)) -> Result<Self::Request, Error> {
-        match h.protocol {
-            None => {
-                // Update the TcpAccept target using a parsed transport-header.
-                t.target_addr = (t.target_addr.ip(), h.port).into();
-                Ok(svc::Either::A(t.into()))
-            }
-            Some(p) => {
-                let v = match p {
-                    transport_header::SessionProtocol::Http1 => http::Version::Http1,
-                    transport_header::SessionProtocol::Http2 => http::Version::H2,
-                };
-                Ok(svc::Either::B((v, t)))
-            }
+// === impl ClientInfo ===
+
+impl Param<transport::labels::Key> for ClientInfo {
+    fn param(&self) -> transport::labels::Key {
+        transport::labels::Key::accept(
+            transport::labels::Direction::In,
+            Conditional::Some(tls::ServerTls::Established {
+                client_id: Some(self.client_id.clone()),
+                negotiated_protocol: None, // unused
+            }),
+            self.local_addr,
+        )
+    }
+}
+
+// === impl HttpClientInfo ===
+
+impl Param<http::normalize_uri::DefaultAuthority> for HttpClientInfo {
+    fn param(&self) -> http::normalize_uri::DefaultAuthority {
+        http::normalize_uri::DefaultAuthority(self.target.clone().map(|t| {
+            http::uri::Authority::from_str(&t.to_string()).expect("Address must be a valid URI")
```

Comment:
should this use `NameAddr::as_http_authority`? https://github.com/linkerd/linkerd2-proxy/blob/73fb0ff5e109b37c80498192a92adf7541edd1f3/linkerd/addr/src/lib.rs#L210-L217